src/services/review/RepoCheckerService.ts

```diff
@@ -346,24 +407,103 @@ export default class RepoCheckerService {
         .commit("site checker logs")
         .push(),
       (error) => new SiteCheckerError(`${error}`)
-    )
+    ).map(() => errors)
+  }
+
+  // adapted from https://stackoverflow.com/questions/56427009/how-to-return-papa-parsed-csv-via-promise-async-await
+  generateRepoErrorsFromCsv = (repo: string): Promise<RepoError[]> => {
+    const filePath = path.join(SITE_CHECKER_REPO_PATH, repo, "log.csv")
+    const file = fs.createReadStream(filePath)
+    return new Promise((resolve, reject) => {
+      Papa.parse(file, {
+        header: true,
+        complete(results) {
+          // validate the csv
+          if (!results.data.every(isRepoError)) {
+            reject(new Error("Invalid CSV"))
+          }
+          resolve(results.data as RepoError[])
+        },
+        error(err) {
+          reject(err)
+        },
+      })
+    })
+  }
+
+  readRepoErrors = (
+    repo: string
+  ): ResultAsync<BrokenLinkErrorDto, SiteCheckerError> => {
+    const filePath = path.join(SITE_CHECKER_REPO_PATH, repo, "log.csv")
+    if (!fs.existsSync(filePath)) {
+      // start the checking of the repo async
+      return this.checkRepo(repo)
+    }
+
+    return this.isCurrentlyLocked(repo)
+      .andThen(() => okAsync<BrokenLinkErrorDto>({ status: "loading" }))
+      .orElse(() =>
+        ResultAsync.fromPromise(
+          this.generateRepoErrorsFromCsv(repo),
+          (error) => {
+            logger.error(`Error reading csv for repo ${repo}: ${error}`)
+            return new SiteCheckerError(
+              `Error reading csv for repo ${repo}: ${error}`
+            )
+          }
+        ).andThen((errors) =>
+          okAsync<BrokenLinkErrorDto>({
+            errors,
+            status: "success",
+          })
+        )
+      )
   }
 
-  checkRepo(repo: string): ResultAsync<undefined, SiteCheckerError> {
+  checkRepo(repo: string): ResultAsync<BrokenLinkErrorDto, SiteCheckerError> {
     /**
      * To allow for an easy running of the script, we can temporarily clone
      * the repo to the EFS
      */
-
     let repoExists = true
+    return this.isCurrentlyLocked(repo)
+      .andThen(() => okAsync<BrokenLinkErrorDto>({ status: "loading" }))
+      .orElse(() => {
+        // this process can run async, so we can just return the loading state early
+        this.createLock(repo)
```

Comment:
we are not awaiting here because we want to do this async right?

Reply:
So if there are errors during this async run, it will just fail?

Reply:
actually not wrong uh in this case will fail silently

Reply:
but users are able to retry, which is i think more impt here than showing error state

Reply:
> but users are able to retry, which is i think more impt here than showing error state
So whats' the user flow? 
They start the run but if it fails, they just see a empty state? Or do they know if there is an error and they need to retry?

Reply:
on error we are redirecting over to the dashboard